dom5.ts

```diff
@@ -605,6 +605,35 @@ export function remove(node: Node) {
   node.parentNode = undefined;
 }
 
+/**
+ * Removes a node and places its children in its place.  If the node
+ * has no parent, the operation is impossible and no action takes place.
+ */
+export function removeNodeSaveChildren(node: Node) {
+  // We can't save the children if there's no parent node to provide
+  // for them.
+  const fosterParent = node.parentNode;
+  if (!fosterParent) {
+    return;
+  }
+  const children = (node.childNodes || []).slice();
+  for (const child of children) {
+    insertBefore(node.parentNode!, node, child);
+  }
+  remove(node);
+}
+
+/**
+ * When parse5 parses an HTML document, it injects a few html tags that
+ * (html, head and body) if they are missing.  This function removes
+ * nodes from the AST which have no location info, so it requires that the
+ * `parse5.parse` be used with the `locationInfo` option of `true`.
+ */
+export function removeFakeNodes(ast: Node) {
+  const injectedNodes = queryAll(ast, (node) => !Boolean(node.__location));
```

Comment:
I think there are several cleanup functions we may want, and ultimately doing essentially `serialize(mutate(parse(source)))` might not be the right process anyway.

Currently we have two uses for removing nodes: 1) remove bloat 2) warning handing

For removing bloat we mostly care about `html`, `head`, `body`, but also maybe added closing tags - which we won't be able to remove from the AST this way. If parse5 keeps location info on closing tags, then what we'd need is a custom serialize function that can omit all parts without location.

Otherwise we might want to consider a string patching approach:

```typescript
const source;
const ast = parse(source);
const newAst = mutate(ast);
const newSource = patch(source, ast, newAst);
```

Ok, but back to this, given our current file mutation, I think we want a function that doesn't recurse down the whole tree and just removes the few tags we care about.